src/Decimal.php

```diff
@@ -273,7 +254,7 @@ public function add($value, ?int $scale = null)
      *
      * @return int
      */
-    protected function resultScale($a, $b, ?int $scale = null): int
+    protected function resultScale(Decimal $a, Decimal $b, ?int $scale = null): int
```

Comment:
We cannot change method signatures for public methods outside of a new major.
Usually the same also applies to internal methods, since they could be extended as such

Reply:
Yes, you're right.